girder-dandi-archive/girder_dandi_archive/util.py

```diff
@@ -0,0 +1,38 @@
+import re
+
+from girder.models.collection import Collection
+from girder.utility import setting_utilities
+from girder.exceptions import ValidationException
+
+DANDISET_ID_COUNTER = "dandi.id_counter"
+DANDISET_ID_LENGTH = 6
+DANDI_STAGING_COLLECTION_NAME = "Dandiset Staging"
+
+dandiset_id_pattern = r"^\d{6}$"
+
+
+@setting_utilities.validator(DANDISET_ID_COUNTER)
+def _validate(doc):
+    if len(str(doc["value"])) > DANDISET_ID_LENGTH:
+        raise ValidationException("Dandiset ID limit exceeded", "value")
```

Comment:
If this happens, it's a huge problem to migrate, so ensure that you'll never have more than one million IDs issued. Ever.